netket/machine/abstract_machine.py

```diff
@@ -65,6 +65,33 @@ def der_log(self, x, out=None):
             """
         return NotImplementedError
 
+    def to_array(self, normalize=True, b_size=10000):
+        if self.hilbert.is_indexable:
+            all_psis = _np.zeros(self.hilbert.n_states, dtype=_np.complex128)
+            batch_states = _np.zeros((b_size, self.hilbert.size))
+            it = self.hilbert.states().__iter__()
+            for i in range(self.hilbert.n_states // b_size + 1):
+                for j in range(b_size):
+                    try:
```

Comment:
I think it would be more efficient to substitute this inner for loop with a list comprehension or similar

Reply:
actually it should be possible to change this for loop with an itertool step entirely

Reply:
if you mean something like `batch_states = np.array([next(it) for _ in range(b_size)])`, this does actually decrease the speed of the function compared to the explicit for loop, possibly due tot the conversion between python and numpy arrays.

Reply:
you can indeed write something along the lines `batch_states = np.array(list(itertools.islice(it, b_size)))`.  I checked the speed and there is not much of a difference.  For the test case I tried (Torch machine on Hilbert space of 20 spins) I get 2.53 s ± 84.9 ms for the current commit and 2.7 s ± 73.2 ms for the loop-less implementation.